src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -26,7 +26,7 @@ public static partial class Marshal
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern int SizeOfHelper(Type t, bool throwIfNotMarshalable);
 
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2006:UnrecognizedReflectionPattern",
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
```

Comment:
probably not a huge deal, but do we have a better string than `UnrecognizedReflectionPattern` now? It sort of made sense when IL2006 was used, but now we are using 30 differnet codes, all with "UnrecognizedReflectionPattern"

Reply:
@mateoatr does this string have any particular meaning? It looks to me like you could put anything here and the linker would treat it the same. I agree it'd be nice to come up with more specific names for these.

Reply:
My understanding is that linker doesn't actually look at that string, so it should be possible to use really anything. @mateoatr  would know for sure.

Reply:
Feel free to not address this in this PR to unblock you for 5.0. But it would be nice to get something like this early in 6.0.

It is totally a convenience for the readers - but it is a useful convenience IMO, so I don't need to look up the error code from https://github.com/mono/linker/blob/master/docs/error-codes.md each time.

Reply:
Filed https://github.com/mono/linker/issues/1432 for this

Reply:
Yeah, any string can be used here and it shouldn't affect the linker.